### DISCUSSION TITLE
Typescript Types not allowing setValue on fieldArray Values

### DISCUSSION THREAD

**Discussion by ckeyes88** [react-hook-form/react-hook-form#5622] [Q&A] [answered]
**Describe the bug**
Sorry for another bug report already.

Again, recently updated to 7.8.6. I have form that has a field array which is an array of objects. Inside that field array we have a form input that needs to use `setValue` which is not actually working as expected. Typescript is giving a compilation error that does not allow us to dynamically dial in to the value param.

**To Reproduce**
Just edit the sandbox below to see the typescript error. The code itself seems to work fine.

**Codesandbox link (Required)**
[Sandbox here](https://codesandbox.io/s/react-hook-form-usefieldarray-forked-efy1g)

**Expected behavior**
I would expect the typings to be happy but they seem to not recognize the dynamic string that I used to dial down to the field.

**Screenshots**
[Another GIF](https://share.getcloudapp.com/OAuQWBRQ)

**Additional context**
- Managed to get this to work by doing 
```ts
setValue(`things.${index}.label` as any, "00:00", {shouldValidate: true});
```

**Comment by bluebill1049** [comment, marked as answer]
<img width="883" alt="Screen Shot 2021-06-16 at 8 48 40 am" src="https://user-images.githubusercontent.com/10513364/122133231-a804fe00-ce7f-11eb-8250-7c9fb29643af.png">

Try this above, it's a limitation with TS support.

**Comment by ckeyes88** [comment]
That makes sense. Just wanted to make sure I wasn't missing something. Thanks!

**Reply by bluebill1049** [reply]
np
